Pod/Classes/public/error/Error.swift

```diff
@@ -0,0 +1,150 @@
+//
+//  Error.swift
+//  Pods
+//
+//  Created by Jake Young on 6/14/16.
+//
+
+import Foundation
+
+/// Debugging levels for WistiaKit
+public enum DebuggingLevel {
+    
+    /// Shhh, fust be silent.
+    case None
+    
+    /// Print the high level activity and errors.
+    case Friendly
+    
+    /// Print all the things.
+    case Annoying
+}
+
+internal func printErrorForDebuggingLevel(debuggingLevel debuggingLevel: DebuggingLevel, message: String? = nil, error: NSError? = nil) {
+    
+    switch debuggingLevel {
+        
+    case .None:
+        break
+    case .Friendly:
+        
+        // We print the message and fallthrough to allow .Annoying case to add to things.
+        guard let message = message else { break }
+        print(message)
+        fallthrough
+        
+    case .Annoying:
+        
+        // We unrap the provided erorr and print the nerdy details.
+        guard let error = error else { break }
+        print("Error: \(error.description)), reason: \(error.localizedFailureReason)")
+    }
+    
+}
+
+
+/// The `Error` struct provides a convenience for creating custom WistiaKit NSErrors and Swift ErrorTypes.
+public struct Error {
+    
+    /// The domain used for creating all Wistia errors.
+    public static let Domain = "com.wistia.error"
```

Comment:
@spinosa thanks for inquiring, the Error struct here might make a whole lot more sense if it had convenience methods to generate NSErrors from these Wistia Errors.

Apple has done a lot of work to automatically bridge ErrorType and NSError, so I might be convincing myself this isn't necessary within the current intended focus for WistiaKit.  With ErrorType paired with enums, its a better design to pass messages etc. but while its "bridged" to NSError, you lose any associated information with it. 

In other words, .MyError("All the things failed.") in Swift loses its associated message when sent over to NSError... when really you want to convert the message and store it in the "userInfo" dictionary for NSError.

Basically, domain, code, etc, would be used to generate NSError objects for situations where NSErrors are expected... so maybe its a question of "how much Objective-C interoperability does WistiaKit need to have?"